snapfaas/src/fs/mod.rs

```diff
@@ -300,12 +300,54 @@ pub struct Gate {
     object_id: UID,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub enum HttpVerb { #[default] GET, POST, PUT, DELETE }
+
+impl From<HttpVerb> for reqwest::Method {
+    fn from(verb: HttpVerb) -> Self {
+        match verb {
+            HttpVerb::GET    => reqwest::Method::GET,
+            HttpVerb::POST   => reqwest::Method::POST,
+            HttpVerb::PUT    => reqwest::Method::PUT,
+            HttpVerb::DELETE => reqwest::Method::DELETE,
+        }
+    }
+}
+
+impl From<reqwest::Method> for HttpVerb {
+    fn from(method: reqwest::Method) -> Self {
+        match method {
+            reqwest::Method::GET    => HttpVerb::GET,
+            reqwest::Method::POST   => HttpVerb::POST,
+            reqwest::Method::PUT    => HttpVerb::PUT,
+            reqwest::Method::DELETE => HttpVerb::DELETE,
+            _ => panic!("Request method {} not supported", method)
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct ServiceInfo {
+    pub url: String,
+    pub verb: HttpVerb,
+    pub token: Option<String>,
+    pub headers: HashMap<String, Vec<u8>>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Service {
+    label: Buckle,
```

Comment:
What protects who can create a `Service` with a particular label?

Reply:
Currently, nothing protects who can create a `Service` with a particular label, but when creating a `Service`, the integrity of the current label is baked into the `Service` object as its owner/creator. Reading from or writing to the `Service` object requires checking if it is as trusted as the caller (i.e. `service.owner` implies  `current_label.integrity`)